File: frontend/src/types/media.ts
export interface MediaUploadResponse {
  id: string
  url: string
  file_name: string
  mime_type: string
  file_size: number
  width?: number
  height?: number
}

export interface MediaRead {
  id: string
  uploader_id: string
  url: string
  file_name: string
  mime_type: string
  file_size: number
  width?: number
  height?: number
  source_type: string
  source_id?: string
  is_public: boolean
  created_at: string
  updated_at: string
}

export interface PostAttachmentRead {
  id: string
  post_id: string
  media_id: string
  sort_order: number
  created_at: string
}

export interface AvatarUploadResponse {
  avatar_url: string
}

File: backend/app/schemas/media.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel
from pydantic import Field


class MediaUploadResponse(BaseModel):
    id: UUID
    url: str
    file_name: str
    mime_type: str
    file_size: int
    width: int | None = None
    height: int | None = None


class MediaRead(BaseModel):
    id: UUID
    uploader_id: UUID
    url: str
    file_name: str
    mime_type: str
    file_size: int
    width: int | None = None
    height: int | None = None
    source_type: str
    source_id: UUID | None = None
    is_public: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PostAttachmentCreate(BaseModel):
    media_ids: list[UUID] = Field(..., max_length=20)


class PostAttachmentRead(BaseModel):
    id: UUID
    post_id: UUID
    media_id: UUID
    sort_order: int
    created_at: datetime

    model_config = {"from_attributes": True}


class AvatarUploadResponse(BaseModel):
    avatar_url: str
